packages/diagrams/frontend/sirius-components-diagrams/src/toolbar/Toolbar.tsx

```diff
@@ -133,6 +141,11 @@ export const Toolbar = ({
 
   const classes = useToolbarStyles();
 
+  let autoLayout = true;
+  if (diagram && isDiagramDescription(diagram.metadata.description)) {
+    autoLayout = diagram.metadata.description.autoLayout;
+  }
+
```

Comment:
I think you should let the autoLayout props and let the DiagramRepresentation give you the answer since it will be there that this information will be retrieved.